overseer/examples/minimal-example.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::HashSet;
+use std::time::Duration;
+
+use futures::pending;
+use futures_timer::Delay;
+use kv_log_macro as log;
+
+use overseer::{Overseer, Subsystem, SubsystemContext, SubsystemJob};
+
+struct Subsystem1;
+
+impl Subsystem1 {
+	async fn run(mut ctx: SubsystemContext<usize>)  {
+		loop {
+			match ctx.try_recv().await {
```

Comment:
to me this smells like a `for_each(..)` with the stream is equiv (given having to figure out the addition of `Delay` and `send_msg`

Reply:
you certainly can do a `for_each` but that leaves you to only one flow of execution:
1. wait for next message
2. receive it and do things

but what if we want something more complex like
1. is there any message to receive?
2.1 yes, ok, lets process it then
2.2 no, nothing to receive at the moment, let's do some housekeeping then
3 let's send some messages of our own regardless of what happened in 2)

So something following in, say, AnV scheme can work:
1. any messages to receive?
2. no, ok, let's hold on a sec
3. Ok, by know we should have already received a desired chunk, let's re-send a message requesting it